packages/vite/src/vite.ts

```diff
@@ -214,17 +213,24 @@ export const bundle: NuxtBuilder['bundle'] = async (nuxt) => {
 
   await nuxt.callHook('vite:extend', ctx)
 
-  nuxt.hook('vite:extendConfig', (config) => {
-    const replaceOptions: RollupReplaceOptions = Object.create(null)
-    replaceOptions.preventAssignment = true
+  nuxt.hook('vite:extendConfig', async (config) => {
+    const replaceOptions = Object.create(null)
 
     for (const key in config.define!) {
       if (key.startsWith('import.meta.')) {
         replaceOptions[key] = config.define![key]
       }
     }
 
-    config.plugins!.push(replace(replaceOptions))
+    // @ts-expect-error Rolldown-specific check
+    if(vite.rolldownVersion) {
+      const { replacePlugin } = await import('rolldown/experimental')
```

Comment:
Will be moved over to a stable plugin soon™️ (https://github.com/rolldown/rolldown/issues/5815)